Sources/Components/Klimabrolet/KlimabroletView.swift

```diff
@@ -0,0 +1,177 @@
+    func klimabroletViewDidSelectClose(_ view: KlimabroletView)
+}
+
+public class KlimabroletView: UIView {
+    // MARK: - Public properties
+
+    public var delegate: KlimabroletViewDelegate?
+
+    public var model: KlimabroletViewModel? {
+        didSet {
+            contentView.titleLabel.text = model?.title
+            contentView.subtitleTagView.titleLabel.text = model?.subtitle
+            contentView.bodyTextLabel.text = model?.bodyText
+            contentView.accessoryButton.setTitle(model?.readMoreButtonTitle, for: .normal)
+            actionsView.primaryButton.setTitle(model?.acceptButtonTitle, for: .normal)
+            actionsView.secondaryButton.setTitle(model?.declineButtonTitle, for: .normal)
+        }
+    }
+
+    // MARK: - Private properties
+
+    private var shadowAnimationDuration = 0.12
+    private var contentSizeObservation: NSKeyValueObservation?
+
+    private(set) lazy var closeButton: UIButton = {
+        let button = UIButton(withAutoLayout: true)
+        button.tintColor = .milk
+        button.setImage(UIImage(named: .newClose), for: .normal)
+        button.backgroundColor = UIColor.black.withAlphaComponent(0.6)
+        button.contentEdgeInsets = UIEdgeInsets(all: 6)
+        button.addTarget(self, action: #selector(handleTapOnCloseButton), for: .touchUpInside)
+
+        return button
+    }()
+
+    private lazy var scrollView: UIScrollView = {
+        let scrollView = UIScrollView(withAutoLayout: true)
+        scrollView.bounces = true
+        scrollView.contentInset = UIEdgeInsets(bottom: .largeSpacing)
+        scrollView.delegate = self
+        return scrollView
+    }()
+
+    private lazy var contentView: KlimabroletContentView = {
+        let view = KlimabroletContentView(withAutoLayout: true)
+        view.delegate = self
+        return view
+    }()
+
+    private lazy var actionsView: KlimabroletActionsView = {
+        let view = KlimabroletActionsView(withAutoLayout: true)
+        view.layer.shadowOffset = .zero
+        view.delegate = self
+        return view
+    }()
+
+    // MARK: - Initializers
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setup()
+    }
+
+    public override func layoutSubviews() {
+        super.layoutSubviews()
+        closeButton.layer.cornerRadius = closeButton.bounds.width / 2.0
+    }
+
+    // MARK: - Private methods
+
+    private func setup() {
+        backgroundColor = .milk
+        layer.cornerRadius = 20
+        clipsToBounds = true
+
+        scrollView.addSubview(contentView)
+
+        NSLayoutConstraint.activate([
+            contentView.leadingAnchor.constraint(equalTo: scrollView.leadingAnchor),
+            contentView.trailingAnchor.constraint(equalTo: scrollView.trailingAnchor),
+            contentView.widthAnchor.constraint(equalTo: scrollView.widthAnchor),
```

Comment:
Is it possible to use `contentView.fillInSuperview()` or do you need to set the width constraint as well?

Reply:
sorry i didn't address your comment before.

I do need the width anchor constraint, otherwise the content within the scroll view would look like:

<img width="545" alt="Screenshot 2019-07-31 at 10 09 39" src="https://user-images.githubusercontent.com/167989/62195048-6cd8af00-b37b-11e9-858d-3f783d05f79d.png">

I guess it's because a scrollview and given the `intrinsicContentSize` of the content view, then it can still be wider than the scrollview even though is constrained to leading and trailing

But the rest of the constraints can be defined with `content.fillInSuperView`.